pkg/api/routes.go

```diff
@@ -2204,6 +2211,80 @@ func (rh *RouteHandler) RevokeAPIKey(resp http.ResponseWriter, req *http.Request
 	resp.WriteHeader(http.StatusOK)
 }
 
+// ChangePassword godoc
+// @Summary Change user's password
+// @Description Change user's password. Invalidates all user's other active sessions.
+// @Router  /zot/auth/change_password [post]
+// @Accept  json
+// @Produce json
+// @Param old_password body string true "Old password"
+// @Param new_password body string true "New password"
+// @Success 200 {string} string "password changed"
+// @Failure 500 {string} string "internal server error"
+// @Failure 401 {string} string "unauthorized"
+// @Failure 400 {string} string "bad request"
+// @Failure 403 {string} string "old password is incorrect".
+func (rh *RouteHandler) ChangePassword(resp http.ResponseWriter, req *http.Request) {
+	// leave OPTIONS method
+	if req.Method == http.MethodOptions {
+		return
+	}
+
+	body, err := io.ReadAll(req.Body)
+	if err != nil {
+		rh.c.Log.Error().Msg("failed to read req body")
+		resp.WriteHeader(http.StatusInternalServerError)
+		_, _ = resp.Write([]byte("internal server error"))
+
+		return
+	}
+
+	var reqBody ChangePasswordRequest
+	if err := json.Unmarshal(body, &reqBody); err != nil {
+		rh.c.Log.Error().Msg("failed to unmarshal req body")
+		resp.WriteHeader(http.StatusBadRequest)
+		_, _ = resp.Write([]byte("bad req"))
+
+		return
+	}
+
+	userAc, err := reqCtx.UserAcFromContext(req.Context())
+	if err != nil {
+		return
+	}
+
+	username := userAc.GetUsername()
```

Comment:
What happens if the call does not contain the information needed to return the user?
For example I allow both local user access and anonymous access on the same server, and someone make a call to this API without providing a username.